### ISSUE TITLE
Use Markdown and similar for markup inside segments

### ISSUE THREAD

**Issue by sujato** [suttacentral/suttacentral#1118]
While our standoff markup will enable us to separate the text and markup in most cases, there will be some instances where markup inside the segments is needed. Ultimately this should be handled via standoff properties, but these are not ready yet.

Meanwhile, I propose we adopt a pragmatic strategy for these, relying on Markdown where possible, and similar plain-text conventions where necessary. Use cases include:

- Emphasis of *italic* and **bold**. Used rarely, but it is used.
- Lists inside segments; currently marked with ~, we should use the markdown - instead.
- For foreign words, we could maybe assign `*asterisks*` to `<em>` for emphasis, and `_underscore_` to `<i>` for foreign words. But how to define language?
- For text critical markup, we can use the conventions used in bendall-cv and extend them as needed. https://github.com/sujato/bendall-cv

**Comment by firepick1** [comment]
While implementing search, it's become apparent that markup degrades the search experience. It degrades the search experience because "what you see is not what you can find". Even simple punctuation has this effect. For example, one can easily search for `blue, yellow` but  not find that with `blue yellow`. And searching for `the [link]` now requires some application acrobatics. Embedded markup is invisible to the end user but not to the search or voice engines. As markup becomes richer, search becomes more quirky and brittle. Standoff properties neatly avoid this conundrum. Another solution is to develop a performant "unmarkdown an-Nilakkhana" module that strips markdown. Then we can regenerate sc-voice/sutta periodically to have a stripped version of supported scriptures.

**Comment by firepick1** [comment]
Yes. It is normal to strip punctuation and that is easily done with a simple regular expression. Oddly the regular syntax of html makes also it easy to strip with a single regular expression. As I was reading your post on md features, my face got pensive and I realized that md is a little trickier. We actually need a variant of the md engine itself to strip md.  Don't worry, though, this is more a note to self about work to be done.
